### PR TITLE
Cleanup Config.py

### PR DESCRIPTION
**Pull request by milliams**
This removes a bunch of unused and unnecessary code from Config as a first step towards making it more easily loadable and un-loadable.

The biggest change in here is that `getConfig()` will no longer *create* a config section if it doesn't find one. This is possible since we are now explicit about using `makeConfig()` in our `__init__.py` files.

### CODE REVIEW

**Comment by rob-c** [comment]
Can we bump this to 6.1.22? If we're aiming for a first release Monday then this seems like it could break a plugin potentially

File: python/Ganga/Utility/Config/Config.py [hunk 1]
```diff
@@ -932,11 +877,8 @@ def setSessionValue(config_name, option_name, value):
 
     # put value in the buffer, it will be removed from the buffer when option
     # is added
-    allConfigFileValues.setdefault(config_name, {})
-    allConfigFileValues[config_name][option_name] = value
-
-
-_configured = False
+    unknownConfigFileValues.setdefault(config_name, {})
```
Comment (alexanderrichards) [inline python/Ganga/Utility/Config/Config.py]: Worth using a defaultdict?

Comment (milliams) [inline python/Ganga/Utility/Config/Config.py]: I'd missed that. Yes, it probably is worth it.

**Comment by mesmith75** [comment]
Do we want this in the next release - it looks like there are conflicts to be resolved..

**Comment by milliams** [comment]
I have one or two small tweaks to make to this so I will defer it.

**Comment by rob-c** [comment]
Is this PR best against 6.2.1 or 6.3.0?

**Comment by milliams** [comment]
I'm not sure whether it would count as a feature or a bugfix. It doesn't have any externally visible changes so is not a feature but at the same time it is not fixing any known bug. It should be safe and on the whole has only been removing unused code so I don't foresee any problems. I would err on the side of 6.2.1 but I'm happy to have it deferred.

**Comment by rob-c** [comment]
@mesmith75 I think I'm happy to make this your call, should we include this in 6.2.1 or 6.3.0?

**Comment by mesmith75** [comment]
Let's put this into 6.2.1.

**Comment by mesmith75** [comment]
If there are no objections I'll merge this.

**Pull request merged**